### ISSUE TITLE
Keylogger miss some keys combination

### ISSUE THREAD

**Issue by DragonzMaster** [MaxXor/xRAT#243]
If I Hold Control[Ctrl] Key and press 'a' for example it logged and while holding Ctrl i pressed another letter 'c' for example it is logged like :
[Control + A]
instead of :
[Control + A] [Control + C]

------------------
the problem occurs also when i press Ctrl key and for example Up arrow and will holding Ctrl I pressed Down arrow  it is logged like :
[Control + Up] [Down]
Instead of :
[Control + Up] [Control + Down]

**Comment by d3agle** [comment]
This is because of how the keys are processed.  The keys are logged on a KeyUp event, so when you press Ctrl + A and you are still holding control key down, it will clear the keys when you let up on 'A', so Control won't be recorded the next time you press 'C' (Unless you let up off of control).

One way to fix this would be to record the keystates of modifier keys with GetKeyState/GetAsyncKeyState functions, when the keys are being cleared from the pressedkeys lists

**Comment by d3agle** [comment]
Now that I think about it I'm not sure this would be easy to do, as far as logging keys accurately, given the logic already implemented.

**Comment by yankejustin** [comment]
@d3agle I've come to realize that keylogging is never easy! hahaha
There are a few solutions I can think of. Maybe we should restrict the clearing of modifiers even more.

**Comment by d3agle** [comment]
@yankejustin, Indeed.  How can we restrict the modifiers even more?  I've thought about some solutions but because we are handling the logging of keys in the KeyUp event, we might have to change the way we log the keys altogether.

**Comment by bitterypaul** [comment]
Does the keylogger catch all the keystrokes and combinations

**Comment by yankejustin** [comment]
@bitterypaul Not 10p% sure if this is still an issue.
@d3able Is this issue still relevant?

**Comment by yankejustin** [comment]
100%*
@d3agle
On my phone currently..Please read the above comment. :)

**Comment by d3agle** [comment]
@yankejustin, Yes it is, I took a look at the HotkeySet class and it looks like we might be able to control logging combinations of keys to accomplish this :D

**Comment by d3agle** [comment]
I mean, it's not really an "issue" per say, however it's something that could be looked into a bit more.  I'm not really sure how we would do this to be honest, but I may take a look at it a little later today. 

@yankejustin, How is your registry editor coming along? :D

**Comment by yankejustin** [comment]
@d3agle Good. Currently about to implement the ability to add <code>RegistryKeys</code> to the <code>ListView</code>. After that works, I will transfer the logic over to the Client so it will finally retrieve <code>RegistryKeys</code> from the Client. :)

**Comment by MaxXor** [comment]
@yankejustin I've created one, named `regedit`.
